src/core/src/op/one_hot.cpp

```diff
@@ -45,61 +47,22 @@ void op::v1::OneHot::validate_and_infer_types() {
     NODE_VALIDATION_CHECK(this,
                           on_value_et.compatible(off_value_et),
                           "on_value element type must be compatible with off_value element type.");
-
+    NODE_VALIDATION_CHECK(this,
+                          depth_et.is_integral(),
+                          "'depth' input element type must be an integer (got ",
+                          depth_et,
+                          ").");
```

Comment:
We've already checked that `depth_et.is_integral() || indices_et.is_dynamic()` above. I think we can remove this re-check